source/main.coffee

```diff
@@ -7,149 +7,53 @@ import * as util from "./util.coffee"
 { SourceMap } = util
 export { parse, generate, util }
 
-# Rules that are not cacheable
-# Essentially anything that depends on mutable state in the parser like indents and the rules that depend on them
-# One day this will be better supported by Hera
+import StateCache from "./state-cache.mts"
+
+# Need to no-cache any rule that directly modifies parser state
+# indentation stack, jsx stack, etc.
+
 uncacheable = new Set [
-  "ActualAssignment"
+  # Meta
+  "Debugger"
+  "Init"
+  "Program"
+  "Reset"
+
+  # Indentation
+  "InsertIndent"
```

Comment:
This shouldn't be needed, as it uses but doesn't modify state. I confirmed this locally, but I'm getting a timeout for the test `should sourcemap complex civet files` — perhaps we should increase our 2000ms timeout for slower machines?

Reply:
👍